lib/components/ArtistScreen/artist_screen_content.dart

```diff
@@ -17,66 +17,60 @@ class ArtistScreenContent extends StatefulWidget {
 }
 
 class _ArtistScreenContentState extends State<ArtistScreenContent> {
-  Future<List<BaseItemDto>?>? songs;
-  Future<List<BaseItemDto>?>? albums;
   JellyfinApiHelper jellyfinApiHelper = GetIt.instance<JellyfinApiHelper>();
 
   @override
   Widget build(BuildContext context) {
-    songs ??= jellyfinApiHelper.getItems(
-      parentItem: widget.parent,
-      filters: "Artist=${widget.parent.name}",
-      sortBy: "PlayCount",
-      includeItemTypes: "Audio",
-      isGenres: false,
-    );
+    final futures = Future.wait([
+      jellyfinApiHelper.getItems(
+        parentItem: widget.parent,
+        filters: "Artist=${widget.parent.name}",
+        sortBy: "PlayCount",
+        includeItemTypes: "Audio",
+        isGenres: false,
+      ),
+      jellyfinApiHelper.getItems(
+        parentItem: widget.parent,
+        filters: "Artist=${widget.parent.name}",
+        sortBy: "ProductionYear",
+        includeItemTypes: "MusicAlbum",
+        isGenres: false,
+      )
+    ]);
 
     return FutureBuilder(
-        future: songs,
-        builder: (context, songSnapshot) {
-          var orderedSongs = songSnapshot.data?.reversed.toList() ?? [];
+        future: futures,
+        builder: (context, snapshot) {
+          var songs = snapshot.data?.elementAt(0) ?? [];
+          var albums = snapshot.data?.elementAt(1) ?? [];
```

Comment:
`elementAt()` throws an exception if the index doesn't exist. You want `elementAtOrNull()` instead :)